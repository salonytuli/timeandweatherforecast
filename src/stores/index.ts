import {createStore, applyMiddleware} from 'redux'
import thunkMiddleware from 'redux-thunk'
import {composeWithDevTools} from 'redux-devtools-extension/developmentOnly'
import rootReducer from '../reducers/rootReducer'
import {ForecastState} from '../types'

interface initialStateInterface {
    forecast: ForecastState
}

const initialStore: initialStateInterface = {
    forecast: {
        forecast: [],
    }
};

const composeEnhancers = composeWithDevTools(applyMiddleware(thunkMiddleware));
export const store = createStore(rootReducer, initialStore, composeEnhancers);
