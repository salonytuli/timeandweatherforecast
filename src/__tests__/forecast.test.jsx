import React from 'react';
import {render} from '@testing-library/react';
import Forecast from '../components/forecast';
import {Provider} from 'react-redux';
import {store} from "../stores";

describe("forecast screen", () => {
    it("should match snapshot", () => {
        const component = render(<Provider store={store}><Forecast/></Provider>);
        expect(component.toJSON).toMatchSnapshot();
    })
});