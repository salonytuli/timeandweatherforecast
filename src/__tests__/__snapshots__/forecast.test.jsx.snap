// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`forecast screen should match snapshot 1`] = `undefined`;

exports[`forecast should match snapshot 1`] = `undefined`;
